### ISSUE TITLE
Audio event classification sample

### ISSUE THREAD

**Issue by khanhlvg** [ml-gde/e2e-tflite-tutorials#32]
* Goal: Build a sample app for [Yamnet](https://github.com/tensorflow/models/tree/master/research/audioset/yamnet), a pretrained model that can recognize many different kinds of audio event.
* References:
  * [Pretrained model](https://tfhub.dev/google/lite-model/yamnet/tflite/1)
  * [TFLite sound classification Android sample](https://github.com/Sa-ryong/examples/tree/master/lite/examples/sound_classification/android)
  * iOS sample coming soon

**Comment by sayakpaul** [comment]
@khanhlvg does this indicate we can now convert an audio classification model trained using Teachable Machine?

**Comment by farmaker47** [comment]
@khanhlvg 
I am definetely interested about that!
Started reading all information...

**Comment by farmaker47** [comment]
I made a colab notebook with executable code blocks and the ability to hear and try different audio files
https://colab.research.google.com/drive/1tM9LIcFr5PCggzUqM0BXr07Zokil-Bye?usp=sharing

**Comment by sayakpaul** [comment]
@farmaker47 do you want to create a separate repository for the project? Also, feel free to send a PR by including the link in the README of this repository. We can then scope this project :D

**Comment by farmaker47** [comment]
Yes sure!

I will create one when I start working the android part!

**Comment by farmaker47** [comment]
Hey @khanhlvg and @sayakpaul 
I paste here some significant results from the benchmark tool for the yamnet tflite model.

![yamnet_pixel_cpu_plus_hexagon](https://user-images.githubusercontent.com/26084498/97809560-28bb1500-1c76-11eb-946b-1654b743d0ab.PNG)

**Comment by farmaker47** [comment]
I want your opinion @khanhlvg @sayakpaul about the fact that I get tflite error for this model when I try to use GPU.
There are no results in every phone configuration.
The TF Hub url is [this](https://tfhub.dev/google/lite-model/yamnet/tflite/1) where there is a code snippet on how to use this model.
Do you think that this model is not for GPU and why is that?

**Comment by sayakpaul** [comment]
@farmaker47 it can be that the model is not yet supported by the TFLite GPU delegate or at least some parts of the model are not yet supported. For example, if I recollect correctly `argmax()` operation is not yet supported. 

So, a good homework for you would be to load up the TFLite model in Netron and compare which ops [are not supported by the delegate](https://www.tensorflow.org/lite/performance/gpu#non-supported_models_and_ops).

**Comment by khanhlvg** [comment]
@farmaker47 @sayakpaul It's great to see the progress on this and sorry for the delayed response from my side.

I think there's a good chance that the model isn't compatible with GPU delegate. To find out, you can run the [benchmark tool](https://www.tensorflow.org/lite/performance/measurement#native_benchmark_binary) with the yamnet model and see the output/error message from the tool to find out.

**Comment by farmaker47** [comment]
Updated [colab notebook](https://colab.research.google.com/drive/1tM9LIcFr5PCggzUqM0BXr07Zokil-Bye?usp=sharing) with inference with interpreter with the [tflite model](https://tfhub.dev/google/lite-model/yamnet/tflite/1).

Procedure of collecting sound, transforming it to correct form and infering with the interpreter seems similar with the one that we did with the SPICE model. So now what is left is to combine everything inside the android application!

**Comment by farmaker47** [comment]
I upload the first outputs from our model when the same .wav file is infered:

- Colab notebook TensorFlow model: 9.9126965e-01 2.6559830e-04 1.5967786e-03 ..........
- Colab notebook TF Lite model: 9.91269529e-01 2.65538692e-04 1.59677863e-03 ..........
- Mobile phone TFLite file: 0.9912695, 2.655983E-4, 0.0015967786 ..........
Clearly we have an exact match and everything seems to work fine!!!

Also I have to mention that Log messages inside android studio from the tensorflow library are now more helpful. Watch an example if I intentionally change the output size of the float array:
E/EXCEPTION: java.lang.IllegalArgumentException: Cannot copy from a TensorFlowLite tensor (Identity) with shape [8, 521] to a Java object with shape [4, 521].
Above the log message points the error and the output node name (Identity)

The inference time for a 2 seconds audio file is 100ms... pretty fast don't you think??!! :) :)

**Comment by khanhlvg** [comment]
Thanks George for the update! It looks like you're making good progress :)

> The inference time for a 2 seconds audio file (after benchmarking it, CPU is used with only 2 threads) is 100ms... pretty fast don't you think??!! :) :)

Indeed! Classification model aren't computational expensive.

**Comment by farmaker47** [comment]
Clearly android app outputs the same labels as colab. I think project is going to finish real fast!
![Capture](https://user-images.githubusercontent.com/26084498/100424271-01e5d800-3096-11eb-9206-fb24d0155b64.PNG)

**Comment by sayakpaul** [comment]
@farmaker47 from our previous discussions, I think that the only thing that is remaining here is the blog post. Right?

**Comment by farmaker47** [comment]
Yes Sayak. I have started writing the blog post.

**Comment by sayakpaul** [comment]
@farmaker47 can we close this issue now as you have already released the blog posts?

**Comment by farmaker47** [comment]
Yeah sure!

Στις Σάβ, 12 Δεκ 2020, 4:10 μ.μ. ο χρήστης Sayak Paul <
notifications@github.com> έγραψε:
